File: day2/task1.py
with open('input.txt') as f:
  correct = 0
  for password in map(str, f):
    (policy, character, password) = password.split()
    (policy_min, policy_max) = policy.split('-')
    character = character[:-1]
    if password.count(character) in range(int(policy_min), int(policy_max)+1):
      correct += 1
  print(correct)

File: day2/task1.php
<?php
$correct = 0;
foreach (file('input.txt') as $line) {
  list($policy, $character, $password) = explode(" ", $line);
  list($policy_min, $policy_max) = explode('-', $policy);
  $count = substr_count(trim($password), substr($character, 0, 1));
  if (($policy_min <= $count) && ($count <= $policy_max))
    ++$correct;
}
print($correct);
